File: server/tests/fixtures/code-graph/sample.ts
// Fixture: structural extraction sample for the F2 extractor tests.
import { Widget } from './widget';
import './side-effect';
import lodash from 'lodash';

export interface Greeter {
  greet(name: string): string;
}

export enum Color {
  Red = 'red',
  Green = 'green',
}

export class WidgetBox {
  private widgets: Widget[] = [];

  add(w: Widget): void {
    this.widgets.push(w);
    this.count();
  }

  count(): number {
    return this.widgets.length;
  }
}

export function createBox(): WidgetBox {
  return new WidgetBox();
}

export function describeBox(): string {
  const box = createBox();
  const total = box.count();
  return `${total}`;
}

File: server/tests/fixtures/code-graph/sample.py
"""Fixture: structural extraction sample for the F2 extractor tests."""
import os
from .helper import greet


class Greeter:
    """A greeter."""

    def __init__(self, name):
        self.name = name

    def hello(self):
        return greet(self.name)


def build_greeter(name):
    return Greeter(name)


def main():
    g = build_greeter("world")
    return g.hello()
